src/map/npc.cpp

```diff
@@ -5802,9 +5803,15 @@ TIMER_FUNC(npc_dynamicnpc_removal_timer){
 }
 
 struct npc_data* npc_duplicate_npc_for_player( struct npc_data& nd, struct map_session_data& sd ){
-	if( sd.npc_id_dynamic != 0 ){
-		clif_msg_color( &sd, C_DYNAMICNPC_TWICE, color_table[COLOR_LIGHT_YELLOW] );
-		return nullptr;
+	int src_id = nd.src_id > 0 ? nd.src_id : nd.bl.id;
```

Comment:
What's that new check here? Like if we have a duplication chain?

Reply:
Just in case the user tries to create a dynamic npc from a duplicate npc. It sounds stupid but our system allows it